src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/FormattingOptions2.cs

```diff
@@ -78,8 +78,9 @@ public Provider()
             new(FeatureName, FormattingOptionGroups.NewLine, nameof(InsertFinalNewLine), DocumentFormattingOptions.Default.InsertFinalNewLine,
             storageLocation: EditorConfigStorageLocation.ForBoolOption("insert_final_newline"));
 
-        public static PerLanguageOption2<FormattingOptions2.IndentStyle> SmartIndent { get; } =
-            new(FeatureName, FormattingOptionGroups.IndentationAndSpacing, nameof(SmartIndent), defaultValue: IndentationOptions.DefaultIndentStyle);
+        public static PerLanguageOption2<IndentStyle> SmartIndent { get; } =
+            new(FeatureName, FormattingOptionGroups.IndentationAndSpacing, nameof(SmartIndent), defaultValue: IndentationOptions.DefaultIndentStyle,
+                new RoamingProfileStorageLocation("TextEditor.%LANGUAGE%.Indent Style"));
```

Comment:
I don't understand how roaming storage location is useful here?
This doesn't seem to be exposed in Tools → Options, and it's not in AutomationObject.

So its value will always remain `DefaultIndentStyle` and it is never fetched/persisted through import/export.

Anything I'm missing? I'm curious to know the the user-visible effect of this change if you noticed any.

Reply:
The option is read from global options here: https://sourceroslyn.io/#Microsoft.CodeAnalysis.LanguageServer.Protocol/Features/Options/IndentationOptionsStorage.cs,21

Global options end up calling `VisualStudioSettingsOptionPersister` to read the option from VS settings, but that only happens if it has the storage.

Reply:
![image](https://user-images.githubusercontent.com/60013978/201487841-592f1b1d-c6e7-4f7c-a791-2d2f3d7b27ba.png)

Reply:
Thanks @tmat @arkalyanms for clarifying.